src/Database/Query/Grammars/PostgresGrammar.php

```diff
@@ -1,6 +1,6 @@
 <?php namespace Winter\Storm\Database\Query\Grammars;
 
-use Winter\Storm\Database\QueryBuilder;
+use Illuminate\Database\Query\Builder;
```

Comment:
yes, you cannot change the contract on a method. Overriding the typehint of the parent class throws a fatal error.  See the related commit message https://github.com/wintercms/storm/pull/26/commits/ad9f2eaa26ae4f4cf856d86c4764f2da4b9d42fb

Reply:
@tschallacka 

> Also I was wondering if it could be removed as the methods are identical. I wasn't 100% sure what the reason was for them existing so I opted for keeping them. When in doubt don't touch it heh.

100% 😄

So long as the methods are identical it should be fine to remove. The upsert methods were added because it was only added to 8.x but I needed it available in Winter using 6.x.

Reply:
I'll go ahead and remove it then.

I would suggest that backports in the future be flagged as such by a comment, so when the framework updates it will be easier to find them and to clean them up.